### PR TITLE
lua-openssl: Introduce new package

### PR DESCRIPTION
**Pull request by amnonpaz**
Maintainer: @amnonpaz
Compile tested: ARM
Run tested: ARM

Description:

Add a Lua wrapper library for openssl.

Add patches for:
1. Revise the makefile, so the package can be built under openwrt
2. Remove "lua_isinteger" implementation from the package's "compat"
   code in order to avoid linkage issues - An implementation already
   exists in the target's Lua package.

Signed-off-by: Amnon Paz <pazamnon@gmail.com>

### CODE REVIEW

**Comment by diizzyy** [comment]
Please don't copyright OpenWrt.org unless you're affiliated/under contract.
https://github.com/openwrt/packages/pull/3274#discussion_r81465499 (doesn't apply to LEDE only)

Use tar.xz instead of tar.gz (saves space)

**Comment by amnonpaz** [comment]
Thanks for your input.
I have fixed the issues you raised, can you please review this commit again?

File: lang/lua-openssl/patches/0001-Revise-Makefile.patch [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
++
++PKGC ?= pkg-config
++
++LIBNAME = openssl.so
++LIB_OPTION = -shared
++
++# lua's package config can be under various names
++LUAPKGC := $(shell for pc in lua lua5.1 lua5.2 lua5.3; do \
++		$(PKGC) --exists $$pc && echo $$pc && break; \
++	done)
```
Comment (karlp) [inline lang/lua-openssl/patches/0001-Revise-Makefile.patch]: Are you sure this works?  Is this guaranteed to only use the pkgconfig from the openwrt environment?

File: lang/lua-openssl/patches/0001-Revise-Makefile.patch [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
++
++BUILD_DIR = $(shell pwd)
++
++# LUA include/libraries build flags
++LUA_LIBDIR := $(shell $(PKGC) --variable=libdir $(LUAPKGC))
++LUA_CFLAGS := $(shell $(PKGC) --cflags $(LUAPKGC))
++LUA_LIBS := $(shell $(PKGC) --libs-only-L $(LUAPKGC))
```
Comment (karlp) [inline lang/lua-openssl/patches/0001-Revise-Makefile.patch]: Same concern here.

**Comment by amnonpaz** [comment]
Are there any other issues with this package?

**Comment by hnyman** [comment]
Well, apparently the upstream git repo has MS-DOS linefeeds (CRLF) in Makefile while your patch 0001 has normal LF, so patching it fails in buildbot...

https://downloads.lede-project.org/snapshots/faillogs/arm_cortex-a9/packages/lua-openssl/compile.txt
```
Applying ./patches/0001-Revise-Makefile.patch using plaintext: 
patching file Makefile
Hunk #1 FAILED at 1 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file Makefile.rej
Patch failed!  Please fix ./patches/0001-Revise-Makefile.patch!
```

**Comment by amnonpaz** [comment]
Strange, I have fixed this issue. I'll take another look at it.

**Comment by build000** [comment]
https://github.com/openwrt/packages/commit/d0697b488b6145bd07efef6e8be889bc43673145 - confirm, this patch fixed problem.

**Pull request merged**